fintech-examples/fintech-ui/src/app/bank/services/consent-authorization.service.ts

```diff
@@ -17,42 +17,62 @@ export class ConsentAuthorizationService {
   fromConsent(okOrNotOk: Consent, redirectCode: string) {
     const authId = this.storageService.getRedirectMap().get(redirectCode).authId;
     const xsrfToken = this.storageService.getRedirectMap().get(redirectCode).xsrfToken;
-    console.log('ConsentAuthorizationService.fromConsent: pass auth id:' + authId + ' for ' + okOrNotOk + ' with redirect code '
-      + redirectCode);
-    this.finTechAuthorizationService.fromConsentGET(authId, okOrNotOk, redirectCode, '', xsrfToken,
-      'response').subscribe(resp => {
-      console.log('ConsentAuthorizationService.fromConsent: ' +
-        'server responded. Now delete redirect cookie for redirect code ', redirectCode);
-      this.storageService.resetRedirectCode(redirectCode);
-      const location = resp.headers.get('Location');
+    console.log(
+      'ConsentAuthorizationService.fromConsent: pass auth id:' +
+        authId +
+        ' for ' +
+        okOrNotOk +
+        ' with redirect code ' +
+        redirectCode
+    );
+    this.finTechAuthorizationService
+      .fromConsentGET(authId, okOrNotOk, redirectCode, '', xsrfToken, 'response')
+      .subscribe(resp => {
+        console.log(
+          'ConsentAuthorizationService.fromConsent: ' +
+            'server responded. Now delete redirect cookie for redirect code ',
+          redirectCode
+        );
+        this.storageService.resetRedirectCode(redirectCode);
+        const location = resp.headers.get('Location');
+
+        // this is added to handle url where to forward after redirection
+        // to be removed when issue https://github.com/adorsys/open-banking-gateway/issues/848 is resolved
+        // or Fintech UI refactored
+        if (this.storageService.getUserRedirected) {
```

Comment:
Why is this put in again. Redirect URL is not unique, because multiple redirects can be active at the same time.